cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -676,6 +676,10 @@ public void deleteApplication(String namespaceId, String appId, String versionId
     getApplicationSpecificationTable().delete(fields);
   }
 
+  public void deleteApplication(ApplicationReference appRef) throws IOException {
+    getApplicationSpecificationTable().deleteAll(getNamespaceAndApplicationRange(appRef));
+  }
+
   public void deleteApplicationEditRecord(ApplicationReference appRef) throws IOException {
```

Comment:
should remove this method and add the deletion logic to the new deleteApplication() method